### ISSUE TITLE
Editing multiple layers

### ISSUE THREAD

**Issue by carrbrpoa** [cmv/cmv-app#591]
Hello,

I have the following scenario: two protected services. Whenever I load the application, it asks for credentials (OK) twice (one for each protected service - OK, I suppose). The problem is that if I cancel the login box for one of them (let's say the one that I know I don't have the credentials for edition), the Editor widget just doesn't show the editable feature, so I can't edit anything.
Any suggestions?

**Comment by roemhildtg** [comment]
Is https://github.com/cmv/cmv-app/issues/567 related?

**Comment by carrbrpoa** [comment]
Hey @roemhildtg, good catch! Maybe. The difference is that in [#567](https://github.com/cmv/cmv-app/issues/567), I was interacting with a custom util/widget; now, I'm interacting with Editor Widget.

**Comment by roemhildtg** [comment]
My initial thought is that its not so much a cmv thing but a arcgis js api identity manager thing. afaik, cmv doesn't do the authenticating, that's baked into the api. Same with the editing, I'm pretty sure the editor widget is simply a cmv ui wrapped around the default api editor widget.

**Comment by carrbrpoa** [comment]
Actually, error happens when I do authenticate in the first login dialog, but seems to be a little different:

![image](https://cloud.githubusercontent.com/assets/10212618/18171443/637fe772-7038-11e6-8058-ace2bbf7ba49.png)

**Comment by tmcgee** [comment]
@carrbrpoa what @roemhildtg says is correct. The authentication occurs within the Esri API's IdentityManager. CMV doesn't get involved.

If you do not enter credentials or you enter incorrect credentials, you will not be able to edit the layers. The Esri editor widget is expecting access to all of the `feature` layers for which you have included via `editorLayerInfos`.

**Comment by carrbrpoa** [comment]
Do you guys think it is possible to workaround this behavior in some way?

**Comment by tmcgee** [comment]
@carrbrpoa I want to make sure I am clear about what you are trying to do.

You want to be able to edit some of the features layers (layers for which you have authenticated successfully with that Feature Service) but exclude other layers for which the authentication has failed or the user cancelled. Is that correct?

**Comment by carrbrpoa** [comment]
Yes, that's it. I want to be able to edit at least the layers that I'm successfully authenticated; in the Layers Widget, it is OK to me to show that empty checkbox for non authed layers, but I want to see the ones I authed in Editor Widget, which is not happening, because as I see and you said, the Editor wants access to all the feature layers.

**Comment by carrbrpoa** [comment]
Another case to help to see this problem is to think about different staff that are able to edit different layers but access the same application.

**Comment by tmcgee** [comment]
You can check your layers before passing them to the Esri editor widget. Somewhere around [here](https://github.com/cmv/cmv-app/blob/master/viewer/js/gis/dijit/Editor.js#L36). So you don't have to customize the Editor widget, an alternative is to create your own widget that receives the same editotLayerInfos, validates the layers and then instantiates the Editor widget.

Actually, this reminds of [this prior discussion](https://github.com/cmv/cmv-app/issues/339) which is a potential cmv enhancement. Pull Request anyone? ;)

**Comment by carrbrpoa** [comment]
Thanks for the help @tmcgee, I'll try something later and post the results!

**Comment by tmcgee** [comment]
@carrbrpoa Did you get this resolved?

**Comment by carrbrpoa** [comment]
Hey @tmcgee! Not yet; reading it more carefully, you talk about "validate the layer". Any tips on how to do it?

**Comment by tmcgee** [comment]
@carrbrpoa by validate the layer, I mean check that the layer is available for editing (the user has authenticated successfully for that specific layer). There may be some code in the issue I linked to that could help but I am not sure. If you write the code in a custom editor widget, you only have to validate those layers that are passed to the widget through `editorLayerInfos`.  Hope that helps clarify what I was suggesting.

**Comment by roemhildtg** [comment]
@carrbrpoa You could listen for the esri identity manager `credential-create` and `dialog-cancel` events to determine which layers are loaded. That could potentially solve both this and https://github.com/cmv/cmv-app/issues/567

https://developers.arcgis.com/javascript/3/jsapi/identitymanager-amd.html#event-credential-create

credential-create	
{
  credential: <Credential>
}
Fired when a credential is created.
credentials-destroy		Fired when all credentials are destroyed.

dialog-cancel	
{
  info: <Object>
}
Fired when the user clicks the cancel button on the dialog box widget.
dialog-create
